### PR TITLE
remove unsupported platforms and future parser

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

**Comment by tphoney** [comment]
@hunner @DavidS removing some crud from the acceptance tests. i may have been over zealous. The failure seen with 

rspec ./spec/acceptance/type_spec.rb:6 # type function success types arrays
rspec ./spec/acceptance/type_spec.rb:18 # type function success types strings

Is seen if you check out current master and run the acceptance tests. "Panda Bug" won't reproduce in captivity but will in the wild.

should the future parser be left in ?

File: spec/acceptance/unsupported_spec.rb [hunk 1]
```diff
@@ -1,11 +0,0 @@
-#! /usr/bin/env ruby -S rspec
-require 'spec_helper_acceptance'
-
-describe 'unsupported distributions and OSes', :if => UNSUPPORTED_PLATFORMS.include?(fact('operatingsystem')) do
-  it 'should fail' do
-    pp = <<-EOS
-    class { 'mysql::server': }
```
Comment (hunner) [inline spec/acceptance/unsupported_spec.rb]: Lolwut, how did this even work before?

File: spec/acceptance/values_spec.rb [hunk 2]
```diff
@@ -13,12 +13,7 @@
       $output = values($arg)
       notice(inline_template('<%= @output.sort.inspect %>'))
       EOS
-      if is_future_parser_enabled?
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[1, 2, 3\]/)
-      else
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\["1", "2", "3"\]/)
-      end
-
+			expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[1, 2, 3\]/)
```
Comment (hunner) [inline spec/acceptance/values_spec.rb]: Hard tabs

File: spec/acceptance/zip_spec.rb [hunk 3]
```diff
@@ -10,11 +10,7 @@
       $output = zip($one,$two)
       notice(inline_template('<%= @output.inspect %>'))
       EOS
-      if is_future_parser_enabled?
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\[1, 5\], \[2, 6\], \[3, 7\], \[4, 8\]\]/)
-      else
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\["1", "5"\], \["2", "6"\], \["3", "7"\], \["4", "8"\]\]/)
-      end
+			expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\[1, 5\], \[2, 6\], \[3, 7\], \[4, 8\]\]/)
```
Comment (hunner) [inline spec/acceptance/zip_spec.rb]: Crunch crunch

File: spec/acceptance/zip_spec.rb [hunk 4]
```diff
@@ -23,11 +19,7 @@
       $output = zip($one,$two)
       notice(inline_template('<%= @output.inspect %>'))
       EOS
-      if is_future_parser_enabled?
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\[1, true\], \[2, true\], \["three", false\], \[4, false\]\]/)
-      else
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\["1", true\], \["2", true\], \["three", false\], \["4", false\]\]/)
-      end
+			expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\[1, true\], \[2, true\], \["three", false\], \[4, false\]\]/)
```
Comment (hunner) [inline spec/acceptance/zip_spec.rb]: Chisel chisel

File: spec/acceptance/zip_spec.rb [hunk 5]
```diff
@@ -40,11 +32,7 @@
       $output = zip($one,$two,true)
       notice(inline_template('<%= @output.inspect %>'))
       EOS
-      if is_future_parser_enabled?
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[1, 5, 2, 6, 3, 7, 4, 8\]/)
-      else
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\["1", "5", "2", "6", "3", "7", "4", "8"\]/)
-      end
+			expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[1, 5, 2, 6, 3, 7, 4, 8\]/)
```
Comment (hunner) [inline spec/acceptance/zip_spec.rb]: Snip snip

File: spec/acceptance/zip_spec.rb [hunk 6]
```diff
@@ -54,11 +42,7 @@
       $output = zip($one,$two)
       notice(inline_template('<%= @output.inspect %>'))
       EOS
-      if is_future_parser_enabled?
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\[1, 5\], \[2, 6\]\]/)
-      else
-        expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\["1", "5"\], \["2", "6"\]\]/)
-      end
+			expect(apply_manifest(pp, :catch_failures => true).stdout).to match(/\[\[1, 5\], \[2, 6\]\]/)
```
Comment (hunner) [inline spec/acceptance/zip_spec.rb]: Chip chip

File: spec/acceptance/unsupported_spec.rb (hunk 1 above)
Comment (tphoney) [inline spec/acceptance/unsupported_spec.rb]: erggh

File: spec/acceptance/values_spec.rb (hunk 2 above)
Comment (tphoney) [inline spec/acceptance/values_spec.rb]: fixed

File: spec/acceptance/zip_spec.rb (hunk 3 above)
Comment (tphoney) [inline spec/acceptance/zip_spec.rb]: fixed

File: spec/acceptance/zip_spec.rb (hunk 4 above)
Comment (tphoney) [inline spec/acceptance/zip_spec.rb]: fixed

File: spec/acceptance/zip_spec.rb (hunk 5 above)
Comment (tphoney) [inline spec/acceptance/zip_spec.rb]: fixed

File: spec/acceptance/zip_spec.rb (hunk 6 above)
Comment (tphoney) [inline spec/acceptance/zip_spec.rb]: fixed

File: spec/acceptance/type_spec.rb [hunk 7]
```diff
@@ -1,29 +1,29 @@
 #! /usr/bin/env ruby -S rspec
 require 'spec_helper_acceptance'
 
-describe 'type function', :unless => (UNSUPPORTED_PLATFORMS.include?(fact('operatingsystem')) || is_future_parser_enabled?) do
+describe 'type function' do
```
Comment (DavidS) [inline spec/acceptance/type_spec.rb]: This test can be deleted completely if we don't support puppet3 anymore. See https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/puppet/parser/functions/type.rb#L11

**Pull request merged**